stream.go

```diff
@@ -714,6 +719,10 @@ func (cs *clientStream) RecvMsg(m interface{}) error {
 	err := cs.withRetry(func(a *csAttempt) error {
 		return a.recvMsg(m, recvInfo)
 	}, cs.commitAttemptLocked)
+	// The context is canceled.
+	if err != nil && err.Error() == toRPCErr(context.Canceled).Error() {
```

Comment:
I'm pretty sure a server could inadvertently trick this code into suppressing the trailers from the client:

```go
func ServiceHandler(...) error {
   if err := doAThingThatMonitorsContext(ctx); err != nil {
      stream.SetTrailer(md)
      return err
   }
}
```

The proper fix is most likely in the transport layer.